cli/azd/cmd/env.go

```diff
@@ -150,6 +168,309 @@ func (e *envSetAction) Run(ctx context.Context) (*actions.ActionResult, error) {
+		}
+	}
+
+	kvAccountSelectionIndex, err := e.console.Select(ctx, input.ConsoleOptions{
+		Message:      pickKvAccount,
+		Options:      selectKvAccountOptions,
+		DefaultValue: selectKvAccountOptions[0],
+	})
+	if err != nil {
+		return nil, fmt.Errorf("selecting Key Vault: %w", err)
+	}
+
+	willCreateNewKvAccount := selectKvAccountOptions[kvAccountSelectionIndex] == createNewKvAccountOption
+	if willCreateNewSecret && !willCreateNewKvAccount {
+		// when willCreateNewSecret is true, we added a new option at the beginning of the list
+		// to recover the original kv account name
+		kvAccountSelectionIndex--
+	}
+
+	var kvAccount keyvault.Vault
+	if atLeastOneKvAccountExists {
+		kvAccount = vaultsList[kvAccountSelectionIndex]
+	}
+
+	if willCreateNewKvAccount {
+		location, err := e.prompter.PromptLocation(ctx, subId, "Select the location to create the Key Vault", nil)
+		if err != nil {
+			return nil, fmt.Errorf("prompting for Key Vault location: %w", err)
+		}
+		rg, err := e.prompter.PromptResourceGroupFrom(ctx, subId, location, prompt.PromptResourceGroupFromOptions{
+			DefaultName: "rg-for-my-kv-account",
+		})
+		if err != nil {
+			return nil, fmt.Errorf("prompting for resource group: %w", err)
+		}
+
+		kvAccountName := ""
+		for {
+			kvAccountNameInput, err := e.console.Prompt(ctx, input.ConsoleOptions{
+				Message: "Enter a name for the Key Vault",
+				Help:    "The name must be unique within the subscription and must be between 3 and 24 characters long",
+			})
+			if err != nil {
+				return nil, fmt.Errorf("prompting for Key Vault name: %w", err)
+			}
+			if kvAccountNameInput == "" {
+				e.console.Message(ctx, "Key Vault name cannot be empty")
+				continue
+			}
+			kvAccountName = kvAccountNameInput
+			break
+		}
+
+		e.console.ShowSpinner(ctx, "Creating Key Vault", input.Step)
+		vault, err := e.kvService.CreateVault(ctx, tenantId, subId, rg, location, kvAccountName)
+		e.console.StopSpinner(ctx, "", input.Step)
+		if err != nil {
+			return nil, fmt.Errorf("Error creating Key Vault: %w", err)
+		}
+		kvAccount = vault
+
+		// RBAC role assignment
+		e.console.ShowSpinner(ctx, "Adding Administrator Role", input.Step)
+		principalId, err := azureutil.GetCurrentPrincipalId(ctx, e.userProfileService, tenantId)
+		if err != nil {
+			return nil, fmt.Errorf("getting current principal ID: %w", err)
+		}
+		err = e.entraIdService.CreateRbac(
+			ctx, subId, kvAccount.Id, keyvault.RoleIdKeyVaultAdministrator, principalId)
+		if err != nil {
+			return nil, fmt.Errorf("adding Administrator Role: %w", err)
+		}
+		e.console.StopSpinner(ctx, "", input.Step)
+	}
+
+	var kvSecretName string
+	if willCreateNewSecret {
+		for {
+			kvSecretName, err = e.console.Prompt(ctx, input.ConsoleOptions{
+				Message:      "Enter a name for the Key Vault secret",
+				DefaultValue: strings.ReplaceAll(secretName, "_", "-") + "-kv-secret",
```

Comment:
I'm wondering how you feel about dropping the `-kv-secret` suffix here. I feel that after dropping the suffix, the suggested default would be closer to a value that the user will just press 'enter' and move on. Appending `-kv-secret` seemed a little less idiomatic in mind. I know we do suffix for Azure resources in general, but not for the properties of Azure resources (like secrets).

```suggestion
				DefaultValue: strings.ReplaceAll(secretName, "_", "-"),
```